src/parser/utils.ts

```diff
@@ -1,65 +1,55 @@
-const map = new WeakMap<object, string>();
-
-const TYPES = {
-  ArrayBuffer: '[object ArrayBuffer]',
-  SharedArrayBuffer: '[object SharedArrayBuffer]',
-  Uint8Array: '[object Uint8Array]',
-  BigInt64Array: '[object BigInt64Array]',
-  BigUint64Array: '[object BigUint64Array]',
-  RegExp: '[object RegExp]',
-  Map: '[object Map]',
-  Date: '[object Date]'
-};
-
-/**
- * Retrieves the prototype.toString() of a value.
- * If the value is an object, it will cache the result in a WeakMap for future use.
- */
-function getPrototypeString(value: unknown): string {
-  let str = map.get(value as object);
-
-  if (!str) {
-    str = Object.prototype.toString.call(value);
-    if (value !== null && typeof value === 'object') {
-      map.set(value, str);
-    }
-  }
-  return str;
-}
-
 export function isAnyArrayBuffer(value: unknown): value is ArrayBuffer {
-  const type = getPrototypeString(value);
-  return type === TYPES.ArrayBuffer || type === TYPES.SharedArrayBuffer;
+  return (
+    typeof value === 'object' &&
+    value != null &&
+    Symbol.toStringTag in value &&
+    (value[Symbol.toStringTag] === 'ArrayBuffer' ||
+      value[Symbol.toStringTag] === 'SharedArrayBuffer')
+  );
 }
 
 export function isUint8Array(value: unknown): value is Uint8Array {
-  const type = getPrototypeString(value);
-  return type === TYPES.Uint8Array;
+  return (
+    typeof value === 'object' &&
+    value != null &&
+    Symbol.toStringTag in value &&
+    value[Symbol.toStringTag] === 'Uint8Array'
```

Comment:
Good catch on the WeakMap causing issues, sorry about that 😅 . This looks to be faster than `prototype.toString` and the WeakMap cache 💪